drivers/display/mb_display.c

```diff
@@ -429,25 +426,43 @@ struct mb_display *mb_display_get(void)
 static int mb_display_init(const struct device *dev)
 {
 	ARG_UNUSED(dev);
-
-	display.dev = device_get_binding(DT_LABEL(DT_NODELABEL(gpio0)));
-
-	__ASSERT(dev, "No GPIO device found");
-
-	gpio_pin_configure(display.dev, LED_ROW1_GPIO_PIN, GPIO_OUTPUT);
-	gpio_pin_configure(display.dev, LED_ROW2_GPIO_PIN, GPIO_OUTPUT);
-	gpio_pin_configure(display.dev, LED_ROW3_GPIO_PIN, GPIO_OUTPUT);
-	gpio_pin_configure(display.dev, LED_COL1_GPIO_PIN, GPIO_OUTPUT);
-	gpio_pin_configure(display.dev, LED_COL2_GPIO_PIN, GPIO_OUTPUT);
-	gpio_pin_configure(display.dev, LED_COL3_GPIO_PIN, GPIO_OUTPUT);
-	gpio_pin_configure(display.dev, LED_COL4_GPIO_PIN, GPIO_OUTPUT);
-	gpio_pin_configure(display.dev, LED_COL5_GPIO_PIN, GPIO_OUTPUT);
-	gpio_pin_configure(display.dev, LED_COL6_GPIO_PIN, GPIO_OUTPUT);
-	gpio_pin_configure(display.dev, LED_COL7_GPIO_PIN, GPIO_OUTPUT);
-	gpio_pin_configure(display.dev, LED_COL8_GPIO_PIN, GPIO_OUTPUT);
-	gpio_pin_configure(display.dev, LED_COL9_GPIO_PIN, GPIO_OUTPUT);
-
+#if defined(CONFIG_BOARD_BBC_MICROBIT)
+	display.dev[0] = device_get_binding(DT_LABEL(DT_NODELABEL(gpio0)));
+
+	__ASSERT(dev[0], "No GPIO device found");
+
+	gpio_pin_configure(display.dev[0], LED_PIN(row_1), GPIO_OUTPUT);
```

Comment:
please, take the opportunity to make this driver use `gpio_configure_dt` API (and related DT macros/API)

Reply:
DT is used as much as for the moment possible.